### DISCUSSION TITLE
ESP32 S3 Pinout und Pinmapping für CMT2300

### DISCUSSION THREAD

**Discussion by Agerter** [hoylabs/OpenDTU-OnBattery#1067] [General]
Hallo,

gibt es irgendwo einen Schaltplan wie ich den CMT2300 und ein Display an das ESP32-S3 Board anschließen kann.
Ist auch ein pinmapping für diese Konfiguration irgendwo verfügbar.
Auf github hab ich nur für den 38er ESP ein Schaltplan gefunden.
Vielen Dank

**Comment by schlimmchen** [comment]
Alle GPIOs am ESP32-S3 sind grundsätzlich für alles verwendbar. Die pin_mapping.json sagt dann, welche Pin-Funktion an welchem GPIO angeschlossen ist. Such es dir aus und erstelle deine pin_mapping.json entsprechend.

**Reply by Agerter** [reply]
Danke für die Rückmeldung... hab noch ne Frage wie kann ich das ESP32 S2 Board mit der neuen openDTu software flashen.

- wo finde ich die aktuelle Software für das ESP32 S2 Board.
- Ich hab da mal ein Onlineflasher gesehen oder meinte das zumindest. Mit dem ESP32 Download flash tool bekomme ich das Board nicht geflashed.
- Danke schonmal im Vorrus

**Reply by schlimmchen** [reply]
Hier ist der Webflasher verlinkt: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/WebFlasher_Introduction

Ich nehme an, dass "ESP32 S2" ein Schreibfehler ist.

**Reply by Agerter** [reply]
ja ein schreibfehler ...... danke erstmal... flashen geht aber WLAN geht nicht , wie ich gelesen habe gibts da bei verschieden Versionen Probleme oder muss ich da was beachten ?

**Comment by Agerter** [comment]
Hallo, also ich hab jetzt je 2 ESP32 S3 von Aliexpress und 2on Amazon getestet..Programm via webflasher erfolgreich geflashed aber ich bekomme keine WLAN verbindung... ich bin mit dem Latein am Ende was kann das sein ??

**Reply by Michel83** [reply]
Hi, nach dem flashen kommt ganz sicher kein Access Point zum anmelden ?

**Reply by Agerter** [reply]
mit dem Webflasher hab ich es jetzt geschafft.. weisst du warum diese RGB LED Leuchtet und blinkt wie ne Discokugel wenn ich mich mit dem WR verbinde??

**Reply by Michel83** [reply]
Hi, sind im PinMapping dafür vielleicht pins gesetzt? Mein S3 ist noch nicht in Betrieb daher kann ich das leider nicht sagen.

**Comment by Heiko63** [comment]
Hallo gibt es schon eine funtionierende Pin mapping für den esp32 s3 ?

**Reply by spcqike** [reply]
Es braucht kein allgemeines, funktionierendes Mapping. Such dir aus, was du wo anschließen willst, und definiere es im Mapping entsprechend. 

https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/1067#discussioncomment-9859809

**Comment by Heiko63** [comment]
Jo, ich habe das openDTU geflasht. sobald ich aber die pin mapping hochgeladen und gespeichert habe ist keine Funktion mehr möglich 
[Test 11.JSON](https://github.com/user-attachments/files/17084070/Test.11.JSON)
meine aktuelle mpin mapping - wo ist der Fehler ?

**Comment by schlimmchen** [comment]
Ich hab mir mal die Mühe gemacht und die Datei heruntergeladen...

```json
[
    {
        "name": "OpenDTU HMS",
        "nrf24": {
            "miso": -1,
            "mosi": -1,
            "clk": -1,
            "irq": -1,
            "en": -1,
            "cs": -1
        },
        "cmt": {
            "clk": 12,
            "cs": 11,
            "fcs": 13,
            "sdio": 14,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": false,
            "phy_addr": 0,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "display": {
            "type": 2,
            "data": 21, 
            "clk": 22   
        }
    }
]
```

Was ist denn 22 für ein Pin? Also nach allem, was ich auf die Schnelle finden kann, gibt's den nicht.

**Reply by knopers1** [reply]
Pin 22 ist doch der Clock Anschluss des Displays, 21 ist der Datapin. Oder hab ich die Frage falsch verstanden?

**Comment by Heiko63** [comment]
ja ich denke auch das Pin 21 der Data Pin und 22 der CLK Pin ist . Werde jetzt noch mal meine Mapping ändern und 21 und 22 auf -1 setzen .
Mal schauen was passiert

**Reply by schlimmchen** [reply]
Ich weiß nicht aus welcher Perspektive du die Pins zählst, jedenfalls sind sie im Pin Mapping als `GPIOx` mit `x` = Zahl im Pin Mapping zu verstehen. So stehts auch in der Doku:

![image](https://github.com/user-attachments/assets/ec17cc0c-09af-4350-891c-30128fb75856)

Und GPIO22 gibt es laut Datenblatt am ESP32-S3 nicht. Also gibt es ihn auch nicht an dem Board, das du einsetzt, zum anschließen. Vermutlich hast du etwas falsch verstanden. Sag mir gerne, was du als Pin22 verstehst, dann kommen wir vielleicht weiter.

![image](https://github.com/user-attachments/assets/50d8030b-1bed-4743-8054-3c9fd0582e95)

Hast du das etwa stumpf aus dem ESP32 (non-S3) Pin Mapping kopiert?

**Reply by schlimmchen** [reply]
Ich habe die Doku aktualisiert:

![image](https://github.com/user-attachments/assets/1e494386-4fbe-49cf-80f9-947a135d1233)

Ist es so hinreicht klar? Oder fehlt noch etwas, damit es verständlich ist?

**Comment by Heiko63** [comment]
So, ich habe die Pin mapping soweit abgeändert das ich alle Werte beim Display auf -1 gesetzt habe.

jetzt läuft der esp ohne Probleme

nur ich bekomme keine Werte vom CMT geliefert.

CMT Status zeigt konfiguriert + verbunden

Habe PB 0  CS    auf pin 11
         PB1   FCS  auf pin 10
        PB2    SDIO auf pin 12
        PD1   CLK   auf pin 13
        
        
[Test 12.JSON](https://github.com/user-attachments/files/17085387/Test.12.JSON)

sollte doch richtig sein oder ?

**Reply by schlimmchen** [reply]
> CMT Status zeigt konfiguriert + verbunden

Das sollte hinreichend sein. Warte mal mind. 15 Minute. Es kann dauern, bis dein WR mit der OpenDTU spricht. Und ist dein WR an einer Batterie angeschlossen? Sonst ist ja gerade eh nichts mit Kommunikation, es braucht DC-Spannung dazu.

> Ist es so hinreicht klar? Oder fehlt noch etwas, damit es verständlich ist?

Kann ich da noch Feedback bekommen, bitte?

**Comment by Heiko63** [comment]
Ne habe keine Batterie angeschlossen.

habe mal ein Foto von der SN gemacht 

deutet das auf "Nullen" der auf "O" ?

![DTU](https://github.com/user-attachments/assets/f141f327-4451-402c-bbe2-ff13c0cabd4e)
![20240921_203807](https://github.com/user-attachments/assets/6a788c84-2128-443c-ab86-c1668d93dd31)

da werde ich wohl bis morgen warten müssen.

und klar bekommst du noch ein Feedback

**Reply by schlimmchen** [reply]
Nullen, immer Nullen, das ist kein Text, sondern eine Hexadezimale Seriennummer.

> da werde ich wohl bis morgen warten müssen.

So ist es.

**Comment by Heiko63** [comment]
danke dir erst mal .

melde mich morgen mit einem ( hoffentlich ) erfolgreichen Raport.

**Comment by Heiko63** [comment]
Guten Morgen !
nachdem ich den ESP heute morgen neu gestartet habe läuft alles super !!
werde bei Gelegenheit noch ein paar Bilder von der Verdrahtung einstellen.
Danke für die schnelle und sehr hilfreiche Unterstützung !!

**Comment by Heiko63** [comment]
so wie auf meinen Platinen
cmt 2300A -  esp32s3 WROOM-1
PB5            -  12
PB6            -  14
PB7            -  13
P00            -  11
 
 verdrahtet - funktioniert mit der Pin Mapping 
 
[Test 12.JSON](https://github.com/user-attachments/files/17090745/Test.12.JSON)

![20240922_192106](https://github.com/user-attachments/assets/7548f0c6-85e4-40af-828c-defce46f3430)
![20240922_192312](https://github.com/user-attachments/assets/e14da741-90f8-49fd-be2e-63318ae55ffa)

**Reply by speddy2222** [reply]
Danke, dein Tipp hat mir geholfen.

**Comment by Heiko63** [comment]
hätte noch eine Idee.
gibt es eine Möglichkeit einen Ausgang zu konfigurieren sobald der " Nachtmodus " aktiv ist ?
und/oder das auch noch am Display anzeigen ?
die Software schaltet zwar die Anzeige-Farbe um - schön wäre da aber auch ein Text
nice to have !!

**Reply by Tpaydt** [reply]
Was meinst du mit die Software schaltet die Farbe um? Vom Display?

Das Display wird doch komplett abgeschaltet wenn der WR offline ist

**Comment by Heiko63** [comment]
Ich meine die Zeile in der Live Ansicht

**Comment by Heiko63** [comment]
![grafik](https://github.com/user-attachments/assets/af3fc76e-da5d-4423-bdcb-12b40f78bbfb)
diese Zeile meinte ich

**Reply by Tpaydt** [reply]
Okay verstehe

**Comment by Heiko63** [comment]
Hallo zusammen,
wollte jetzt noch ein Display OLED Display SH1106 anschliessen.
habe in der Pinmapping des esp32s3 Data auf 9 und clock auf 10 gesetzt.
leider bleibt das Display dunkel.
was mache ich falsch ?
[pin-mapping-mitDisplay.JSON](https://github.com/user-attachments/files/17270765/pin-mapping-mitDisplay.JSON)

**Reply by Solarteur** [reply]
Im Pin-Mapping hast du für SH1106 Type "2" eingetragen.
Gemäß Wiki ist SH1106 Type "3".
![grafik](https://github.com/user-attachments/assets/ba784f61-bd19-4af2-8000-fe867ef3d32d)
Das könnte der Fehler sein.

**Comment by Heiko63** [comment]
Es scheint egal zu sein ob Type 2 oder 3 eingetragen ist.
Hatte das Display wie in der Anleitung auf 5V  angeschlossen.
Leitung jetzt auf 3.3V und schon läufts.

![20241007_194458](https://github.com/user-attachments/assets/3378fd18-64a3-4448-804c-637b02270af5)

**Comment by Heiko63** [comment]
Hallo;
hat jemand schon ein Gehäuse für ESP32s3 mit cmt 2300A und 1,3 Zoll Display konstruiert ?
Alles was ich gefunden habe ist mit dem ESP32 und oder dem kleinen Display.....

**Reply by nukleuz80** [reply]
https://makerworld.com/de/more-models/opendtu-3d-print-model-download?orderBy=score

**Reply by Sorbat** [reply]
> https://makerworld.com/de/more-models/opendtu-3d-print-model-download?orderBy=score

Den link finde ich nicht hilfreich. 
Gefragt wurde nach einem Gehäuse für einen S3. Der link verweist auf eine makerworld Suche nach openDTU.
Das schafft wohl auch jeder selbst.

**Comment by Heiko63** [comment]
Hallo zusammen,
welches Programm genutzt ihr zum bearbeiten der bin ?
wollte gern ich noch ein paar GPIO´s ansteuern/einlesen
z.B  es wird eingespeist , Überwachung der Batteriespannung, etc 
Danke für ein paar Info´s

**Comment by nukleuz80** [comment]
Vielen Dank für diesen Beitrag - er hat mir sehr geholfen! Bei mir hat sich die openDTU erst am nächsten Morgen mit dem WR verbunden. Zur Info.

**Comment by Starfoxfs** [comment]
Hi,
ich hab das gleiche Problem mit einem HMS-2000-4T und einem CMT2300A, bekomm einfach keinen Empfang.


Mein Board ist das ESP32-S3 Devkit C1 mit dem Dual USB oben:

https://i0.wp.com/randomnerdtutorials.com/wp-content/uploads/2024/09/ESP32-S3-pinout.jpg?quality=100&strip=all&ssl=1

ESP32S3 Pin 11 -> CS (CMT2300)
ESP32S3 Pin 12 -> CLK (CMT2300)
ESP32S3 Pin 13 -> FCS (CMT2300)
ESP32S3 Pin 14 -> SDIO (CMT2300)
ESP32S3 Pin 2 -> GPIO2 (CMT2300)
ESP32S3 Pin 3 -> GPIO3 (CMT2300)

ESP32S3 3v3 -> 3v3 (CMT2300)
ESP32S3 GND -> GND (CMT2300)


Mein Pin Mapping:

` {
        "name": "OpenDTU Fusion v2 with CMT2300A and SSD1306 Display",
        "nrf24": {
            "miso": -1,
            "mosi": -1,
            "clk": -1,
            "irq": -1,
            "en": -1,
            "cs": -1
        },
        "cmt": {
            "clk": 12,
            "cs": 11,
            "fcs": 13,
            "sdio": 14,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": false,
            "phy_addr": -1,
            "power": -1,
            "mdc": -1,
            "mdio": -1,
            "type": 0,
            "clk_mode": 0
        },
        "led": {
            "led0": -1,
            "led1": -1
        },
        "display": {
            "type": 3,
            "data": 8,
            "clk": 9
        }
    }`
<img width="1920" height="869" alt="Bildschirmfoto vom 2026-06-13 16-12-58" src="https://github.com/user-attachments/assets/6375b3b3-c479-497a-b34b-4e3c6673c7ec" />
<img width="1280" height="847" alt="Bildschirmfoto vom 2026-06-13 16-13-24" src="https://github.com/user-attachments/assets/294eae06-c057-480d-b746-533e52dc8133" />

<img width="1280" height="955" alt="Bildschirmfoto vom 2026-06-13 16-14-02" src="https://github.com/user-attachments/assets/36a3368c-c802-4801-bf3c-ed6db61d8ebb" />
<img width="1280" height="801" alt="Bildschirmfoto vom 2026-06-13 16-17-15" src="https://github.com/user-attachments/assets/e15e7cfd-1118-4300-af0e-347ad019dbc0" />
